lite-rpc/src/cli.rs

```diff
@@ -1,42 +1,134 @@
 use clap::Parser;
+use dotenv::dotenv;
+use tokio::io::AsyncReadExt;
 
 #[derive(Parser, Debug, Clone)]
 #[command(author, version, about, long_about = None)]
 pub struct Args {
-    #[arg(short, long, default_value_t = String::from(DEFAULT_RPC_ADDR))]
+    /// config.json
+    #[arg(short, long)]
+    pub config: Option<String>,
+    /// identity keypair
+    #[arg(short = 'k', long)]
+    pub identity_keypair: Option<String>,
+}
+
+#[derive(Debug, serde::Deserialize)]
+pub struct Config {
+    #[serde(default = "Config::default_rpc_addr")]
     pub rpc_addr: String,
-    #[arg(short, long, default_value_t = String::from(DEFAULT_WS_ADDR))]
+    #[serde(default = "Config::default_ws_addr")]
     pub ws_addr: String,
-    #[arg(short = 'l', long, default_value_t = String::from("[::]:8890"))]
+    #[serde(default = "Config::default_lite_rpc_http_addr")]
     pub lite_rpc_http_addr: String,
-    #[arg(short = 's', long, default_value_t = String::from("[::]:8891"))]
+    #[serde(default = "Config::default_lite_rpc_ws_addr")]
     pub lite_rpc_ws_addr: String,
-    /// tpu fanout
-    #[arg(short = 'f', long, default_value_t = DEFAULT_FANOUT_SIZE) ]
+    #[serde(default = "Config::default_fanout_size")]
     pub fanout_size: u64,
-    /// enable logging to postgres
-    #[arg(short = 'p', long)]
+    #[serde(default)]
     pub enable_postgres: bool,
-    /// enable metrics to prometheus at addr
-    #[arg(short = 'm', long, default_value_t = String::from("[::]:9091"))]
+    #[serde(default)]
+    pub identity_keypair: Option<String>,
+    #[serde(default = "Config::default_prometheus_addr")]
     pub prometheus_addr: String,
-    #[arg(short = 'k', long, default_value_t = String::new())]
-    pub identity_keypair: String,
-    #[arg(long, default_value_t = MAX_RETRIES)]
+    #[serde(default = "Config::default_maximum_retries_per_tx")]
     pub maximum_retries_per_tx: usize,
-    #[arg(long, default_value_t = DEFAULT_RETRY_TIMEOUT)]
+    #[serde(default = "Config::default_transaction_retry_after_secs")]
     pub transaction_retry_after_secs: u64,
-    #[arg(long)]
+    #[serde(default)]
     pub quic_proxy_addr: Option<String>,
-    #[arg(short = 'g', long)]
+    #[serde(default)]
     pub use_grpc: bool,
-    /// grpc address
-    #[arg(long, default_value_t = String::from(DEFAULT_GRPC_ADDR))]
+    #[serde(default = "Config::default_grpc_addr")]
     pub grpc_addr: String,
-    #[arg(long)]
+    #[serde(default)]
     pub grpc_x_token: Option<String>,
 }
+
+impl Config {
+    pub async fn load() -> anyhow::Result<Self> {
+        dotenv().ok();
+
+        let args = Args::parse();
+
+        let config_path = if let Some(config) = &args.config {
+            config
+        } else {
+            let default_config_path = "config.json";
+
+            // check if config.json exists in current directory
+            if tokio::fs::metadata(default_config_path).await.is_err() {
+                return Ok(serde_json::from_str("{}").unwrap());
```

Comment:
Should return error, config file should exist.

Reply:
There are defaults populated so in case there is no config passed in the cli and no default config 'config.json' then use the defaults